### PR TITLE
MEN-4948: Document the Mender monitor installation instructions

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+title: Install the Mender monitor add-on
+taxonomy:
+    category: docs
+    label: tutorial
+---
+
+!!! Note: Mender monitor is a paid add-on of the Mender project, and does
+!!! require a hosted-mender user login.
```
Comment (lluiscampos) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:9]: @eysteins Should we document here how to download the files for Mender Enterprise on-prem? In the case of `mender-binary-delta`, we have instructions for both flows in Mender Hub, see [here](https://hub.mender.io/t/robust-delta-update-rootfs/1144).

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Download the Mender monitor add-on from
+https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
+and save the tarball in a known location on your local system using your hosted
+Mender username and password:
+
+<!--AUTOVERSION: "/mender-monitor/yocto/%/"/mender-monitor "/mender-monitor_%.tar.gz"/mender-monitor -->
+```bash
+
+    HOSTED_MENDER_EMAIL=<your.email@example.com>
+    curl --fail -u $HOSTED_MENDER_EMAIL -O https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
```
Comment (oleorhagen) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:21]: Note: not the right link (untested)

File: 09.Downloads/docs.md [hunk 3]
```diff
@@ -343,3 +343,32 @@ Please refer to your host Operating System documentation for more details.
...
+The easiest way to install Configure on an existing device is by using the
+Mender APT repository, alternate installation methods are found on the [add-on page for Mender
+monitor](../09.Add-ons/50.Monitor/docs.md).
+
+To install `mender-monitor` using the Mender monitor Debian package, first download it by running:
+
+<!--AUTOVERSION: "/mender-monitor_%-1_all.deb"/mender-monitor "/mender-monitor/debian/%/"/mender-monitor -->
+```bash
+    HOSTED_MENDER_EMAIL=<your.email@example.com>
+    curl --fail -u "$HOSTED_MENDER_EMAIL" -O https://download.mender.io/content/hosted/mender-monitor/debian/master/mender-monitor_master-1_all.deb
```
Comment (oleorhagen) [inline 09.Downloads/docs.md:366]: Note: Not the right link (untested)

**Comment by merlin-northern** [comment]
@eysteins perhaps you should review -- as you usually do with the docs PRs

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+---
+title: Install the Mender monitor add-on
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:2]: suggested change: "Mender monitor" -> "Mender _M_onitor"

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+taxonomy:
+    category: docs
+    label: tutorial
+---
+
+!!! Note: Mender monitor is a paid add-on of the Mender project, and does
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:8]: suggested change: "paid" -> "_commercial_"

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:9]: I am not sure, do we refer to hosted Mender as "hosted-mender"? (I have not checked, but I think I see it for the first time)

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+!!! require a hosted-mender user login.
+
+<!--AUTOVERSION: "/mender-monitor/yocto/%/"/mender-monitor "/mender-monitor_%.tar.gz"/mender-monitor -->
+Download the Mender monitor add-on from
+https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
+and save the tarball in a known location on your local system using your hosted
+Mender username and password:
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:15]: see above, "hosted Mender" looks better.

Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:15]: to "save the tarball using [..] username and password" sounds strange

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+<!--AUTOVERSION: "/mender-monitor/yocto/%/"/mender-monitor "/mender-monitor_%.tar.gz"/mender-monitor -->
+```bash
+
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:19]: do we need this empty line?

File: 09.Add-ons/50.Monitor/10.Installation/docs.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+title: Installation
+taxonomy:
+    category: docs
+---
+
+Mender monitor is a closed source add-on for the Mender project. There are
+multiple ways you can install the Mender monitor add-on into your device image:
+
+* [With Yocto](../../../05.System-updates-Yocto-Project/05.Customize-Mender/docs.md#mender-connect) for the installation in a Yocto Project environment
+* [With a Debian package](../../../09.Downloads/docs.md#remote-terminal-add-on) for the installation from the Debian package
```
Comment (merlin-northern) [inline 09.Add-ons/50.Monitor/10.Installation/docs.md:11]: both ending clauses "for the installation [..]" play nice with the "please consult" but at the top we have "There are multiple ways you can install [..] into your image:"

File: 09.Downloads/docs.md [hunk 9]
```diff
@@ -343,3 +343,32 @@ Please refer to your host Operating System documentation for more details.
...
 [x.x.x_mender-cli-darwin]: https://downloads.mender.io/mender-cli/master/darwin/mender-cli
+
+## Mender monitor
+
+!!! Note: Mender monitor is a paid add-on of the Mender project, and does
+!!! require a hosted-mender user login.
+
+
+Mender offers a monitor add-on (`mender-monitor`) which enables monitoring your
+devices for events and anomalies. configuration. See the [add-on page for Mender
```
Comment (merlin-northern) [inline 09.Downloads/docs.md:354]: " configuration." left over?

File: 09.Downloads/docs.md [hunk 10]
```diff
@@ -343,3 +343,32 @@ Please refer to your host Operating System documentation for more details.
...
+monitor](../09.Add-ons/50.Monitor/docs.md) for more information.
+
+The easiest way to install Configure on an existing device is by using the
+Mender APT repository, alternate installation methods are found on the [add-on page for Mender
```
Comment (merlin-northern) [inline 09.Downloads/docs.md:358]: "are found" -- passive

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md (hunk 1 above)
Comment (estenberg) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:9]: Yes, I think on-premise needs to be documented as well, it's part of our offering after all. :)

Comment (estenberg) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:9]: It's "hosted Mender".

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md (hunk 4 above)
Comment (estenberg) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:2]: Indeed, capital M - it's a name.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md (hunk 5 above)
Comment (estenberg) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:8]: Agree use, "commercial" and "Monitor"
"does require" -> "requires"

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    curl --fail -u $HOSTED_MENDER_EMAIL -O https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
+```
+
+Add the `meta-mender` commercial layer to your build layers:
+
+```bash
+    bitbake-layers add-layer ../sources/meta-mender/meta-mender-commercial
+```
+
+Accept the Mender commercial license by adding to your `local.conf`:
```
Comment (estenberg) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:30]: Minor legal note, but I don't think you can say "Accept the license", this is their choice. Would rephrase a bit.

"To use Mender Monitor, you need to accept its license. If you decide to accept it, add the following to your `local.conf`:

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+```bash
+    IMAGE_INSTALL_append = " mender-monitor"
+```
+
+Which means your `local.conf` should now contain the following lines:
+
+<!--AUTOVERSION: "/mender-monitor_%.tar.gz"/mender-monitor -->
+```bash
+[snip]
```
Comment (estenberg) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:53]: I would skip the snippets above, just show this one and explain quickly each below. That is just one copy-paste and faster.

File: 09.Add-ons/50.Monitor/10.Installation/docs.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+---
+title: Installation
+taxonomy:
+    category: docs
+---
+
+Mender monitor is a closed source add-on for the Mender project. There are
```
Comment (estenberg) [inline 09.Add-ons/50.Monitor/10.Installation/docs.md:7]: Monitor, "commercial"

File: 09.Add-ons/50.Monitor/10.Installation/docs.md (hunk 8 above)
Comment (estenberg) [reply 09.Add-ons/50.Monitor/10.Installation/docs.md:11]: Wrong URL, it's for remote terminal.

File: 09.Downloads/docs.md [hunk 14]
```diff
@@ -343,3 +343,32 @@ Please refer to your host Operating System documentation for more details.
 [x.x.x_mender-cli-linux]: https://downloads.mender.io/mender-cli/master/linux/mender-cli
 <!--AUTOVERSION: "mender-cli/%/"/mender-cli -->
 [x.x.x_mender-cli-darwin]: https://downloads.mender.io/mender-cli/master/darwin/mender-cli
+
+## Mender monitor
```
Comment (estenberg) [inline 09.Downloads/docs.md:347]: Monitor

File: 09.Downloads/docs.md (hunk 10 above)
Comment (estenberg) [reply 09.Downloads/docs.md:358]: Configure -> Monitor

**Review by estenberg — CHANGES_REQUESTED** [review]
Please relocate the Yocto part, it's currently under: 

*  05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor

but the parent page already contains an add-on section you can add it to: https://docs.mender.io/system-updates-yocto-project/customize-mender#add-ons

Otherwise some minor suggestions.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md (hunk 6 above)
Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/02.Mender-monitor/docs.md:15]: ok I do not know what my comment was stranger than the fragment it refers to. maybe just a nitpick: maybe starting with "Using your hosted Mender username and password download .." ? up to you.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 15]
```diff
@@ -251,3 +251,56 @@ do_install_append() {
...
+<!--AUTOVERSION: "/mender-monitor/yocto/%/"/mender-monitor "/mender-monitor_%.tar.gz"/mender-monitor -->
+Download the Mender Monitor add-on from
+https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
+and download the tarball to a known location on your local system using your hosted
+Mender username and password:
+
+<!--AUTOVERSION: "/mender-monitor/yocto/%/"/mender-monitor "/mender-monitor_%.tar.gz"/mender-monitor -->
+```bash
+    HOSTED_MENDER_EMAIL=<your.email@example.com>
+    curl --fail -u $HOSTED_MENDER_EMAIL -O https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
```
Comment (lluiscampos) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:269]: We do not have access to the `mender-monitor` binaries using Hosted Mender user and password, we authentication methods available for `mender-monitor` are:
* JWT for Hosted Mender. The JWT is only available from the GUI, so no direct downloads are possible
* user/password for Mender Enterprise on-prem

Maybe it wasn't obvious for everyone during my rant during SU yesterday (it wasn't my clearer speech and I apologize for that), but this won't work. So we need to re-think what are we going to document.

ping @eysteins

Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:269]: Update,

We had a call today on the topic and concluded that we must fix before release. Being tracked at [MEN-5042](https://tracker.mender.io/browse/MEN-5042).

Comment (oleorhagen) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:269]: @lluiscampos from what I could gather, this can now be done with password everywhere?

Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:269]: @oleorhagen Correct! Now it should work the same way as `mender-binary-delta`: user and pass both for hosted Mender and on-prem Enterprise. Not yet in production, though, so the links can't be tested.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 16]
```diff
@@ -251,3 +251,67 @@ do_install_append() {
 ```
 
 Replace <DIRECTORY-WITH-MENDER-CONNECT-CONF> with the path to the `mender-connect.conf` file, relative to the recipe file.
+
+### Monitor
+
+!!! Note: Mender Monitor is a commercial add-on of the Mender project, and
+!!! requires a hosted Mender user login.
```
Comment (estenberg) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:258]: "hosted"? It can be used by Enterprise on-premise as well..

I would just make this a standard disclaimer like we have for the other add-on packages:

"The Mender Monitor add-on package is required. See the [Mender features page](https://mender.io/plans/features?target=_blank) for an overview of all Mender plans and features."

File: 09.Downloads/docs.md [hunk 17]
```diff
@@ -343,3 +343,44 @@ Please refer to your host Operating System documentation for more details.
 [x.x.x_mender-cli-linux]: https://downloads.mender.io/mender-cli/master/linux/mender-cli
 <!--AUTOVERSION: "mender-cli/%/"/mender-cli -->
 [x.x.x_mender-cli-darwin]: https://downloads.mender.io/mender-cli/master/darwin/mender-cli
+
+## Monitor
+
+!!! Note: Mender Monitor is a commercial add-on of the Mender project, and does
```
Comment (estenberg) [inline 09.Downloads/docs.md:349]: Same here, use a standardized disclaimer. :)

**Review by estenberg — CHANGES_REQUESTED** [review]
Overall looks good, just standardize the disclaimers. Good idea to try to follow the structure of other add-ons to save time and avoid confusion for users.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 18]
```diff
@@ -251,3 +251,66 @@ do_install_append() {
...
 
 Replace <DIRECTORY-WITH-MENDER-CONNECT-CONF> with the path to the `mender-connect.conf` file, relative to the recipe file.
+
+### Monitor
+
+!!! Note: The Mender Monitor add-on package is required. See the [Mender features page](https://mender.io/plans/features?target=_blank) for an overview of all Mender plans and features.
+
+<!--AUTOVERSION: "/mender-monitor/yocto/%/"/mender-monitor "/mender-monitor_%.tar.gz"/mender-monitor -->
+Download the Mender Monitor add-on from
+https://download.mender.io/content/hosted/mender-monitor/yocto/master/mender-monitor_1.0.0.tar.gz
```
Comment (lluiscampos) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:261]: Discussed in person (sorry I didn't review in detail this PR): all these versions need to be `master`, it will be later with the autoversion tool that we will mark 1.0.0
